src/map/script.c

```diff
@@ -1577,6 +1667,74 @@ static const char *parse_line(const char *p)
 	return p;
 }
 
+static const char *parse_syntax_function (const char *p, bool is_public)
+{
+	const char *func_name = script->skip_space(p);
+	p = script->skip_word(func_name);
+
+	if (p == func_name) {
+		disp_error_message("script:parse_syntax_function: function name is missing or invalid", p);
+	}
+
+	const char *p2 = script->skip_space(p);
+
+	if (*p2 == ';') {
+		// function <name> ;
+		// function declaration - just register the name
+		int l = script->add_word(func_name);
+
+		if (script->str_data[l].type == C_NOP) {
+			// register only, if the name was not used by something else
+			script->str_data[l].type = C_USERFUNC;
+		} else if (script->str_data[l].type == C_USERFUNC) {
+			;  // already registered
+		} else {
+			disp_error_message("script:parse_syntax_function: function name is already in use", func_name);
+		}
```

Comment:
To avoid that empty block you could change `== C_USERFUNC` to `!= C_USERFUNC` and show the error message in this block and remove the `else` block completely.
(Having an additional block just to insert a comment brings no benefit in my opinion.)